### PR TITLE
Attribute support

### PR DESCRIPTION
**Pull request by lmars**
So after adding another test, I realised I should probably be starting etcd / discoverd for each test, so I moved that setup into the test.

I also added a `Service::Update` class (to replace using a `Hash` for updates)

WDYT?

### CODE REVIEW

File: test/integration/test_registration.rb [hunk 1]
```diff
@@ -1,17 +1,81 @@
...
+    sleep(0.2)
+
+    @client = Discover::Client.new
+  end
+
+  def teardown
+    sleep(0.2)
+    Process.kill("TERM", @discoverd_pid)
+    Process.kill("TERM", @etcd_pid)
+  end
```
Comment (titanous) [inline test/integration/test_registration.rb:21]: Add `Process.wait`?

**Comment by titanous** [comment]
LGTM.

File: lib/discover.rb [hunk 2]
```diff
@@ -86,9 +87,29 @@ def read_responses
...
     include Celluloid
 
-    def initialize(client, name)
+    class Update < Struct.new(:address, :attributes, :name, :online)
+      def self.from_hash(hash)
+        new *hash.values_at("Addr", "Attrs", "Name", "Online")
+      end
+
+      def online?
+        online == true
```
Comment (titanous) [inline lib/discover.rb:96]: Do we even need the `== true`?

Comment (lmars) [inline lib/discover.rb:96]: I guess not, but I always like to be explicit (e.g. the string "false" should probably be considered false), happy to change though?

Comment (titanous) [inline lib/discover.rb:96]: No problem, leave it.

File: test/integration/test_registration.rb (hunk 1 above)
Comment (lmars) [inline test/integration/test_registration.rb:21]: I stuck a `Process.waitall` in there (see [here](https://github.com/lmars/discoverd-ruby/commit/4f420aeca051d711b0ac207424436d1dd4e78543#diff-4f65a34a1b900b52cc3b4181d4d25da4R18))

**Comment by titanous** [comment]
Good to merge?

**Comment by lmars** [comment]
@titanous sure

**Comment by titanous** [comment]
Thanks!

**Pull request merged**